language/stdlib/modules/validator_config.move

```diff
@@ -1,156 +1,154 @@
+    ///////////////////////////////////////////////////////////////////////////
 
-    // Get Config
-    public fun get_config(addr: address): Config acquires T {
-        *&borrow_global<T>(addr).config
+    public fun publish(signer: &signer) {
+        Transaction::assert(Transaction::sender() == 0xA550C18, 1101);
+        move_to(signer, T {
+            config: Option::none(),
+            operator_account: Option::none(),
+        });
     }
 
-    // Get consensus_pubkey from Config
-    public fun get_consensus_pubkey(config_ref: &Config): vector<u8> {
-        *&config_ref.consensus_pubkey
-    }
-
-    // Returns the address of the entity who is now in charge of managing the config.
-    public fun get_validator_operator_account(addr: address): address acquires T {
-        Option::get_with_default(&borrow_global<T>(addr).delegated_account, addr)
-    }
-
-    // Register the transaction sender as a candidate validator by creating a ValidatorConfig
-    // resource under their account. Note that only one such resource can be
-    // instantiated under an account.
-    public fun register_candidate_validator(
-        consensus_pubkey: vector<u8>,
-        validator_network_signing_pubkey: vector<u8>,
-        validator_network_identity_pubkey: vector<u8>,
-        validator_network_address: vector<u8>,
-        full_node_network_identity_pubkey: vector<u8>,
-        full_node_network_address: vector<u8>) {
-
-        move_to_sender<T>(
-            T {
-                config: Config {
-                    consensus_pubkey: consensus_pubkey,
-                    validator_network_signing_pubkey,
-                    validator_network_identity_pubkey,
-                    validator_network_address,
-                    full_node_network_identity_pubkey,
-                    full_node_network_address,
-                },
-                delegated_account: Option::none()
-            }
-        );
-    }
+    ///////////////////////////////////////////////////////////////////////////
+    // Rotation methods callable by ValidatorConfig::T owner
+    ///////////////////////////////////////////////////////////////////////////
 
-    // If the sender of the transaction has a ValidatorConfig::T resource,
-    // this function will delegate management of the ValidatorConfig::T resource
-    // to a delegated_account.
-    public fun set_delegated_account(delegated_account: address) acquires T {
-        Transaction::assert(LibraAccount::exists(delegated_account), 5);
-        // check delegated address is different from transaction's sender
-        Transaction::assert(delegated_account != Transaction::sender(), 6);
-        let t_ref = borrow_global_mut<T>(Transaction::sender());
-        t_ref.delegated_account = Option::some(delegated_account)
+    // Sets a new operator account, preserving the old config.
+    // If the resource does not exist, creates it.
+    public fun rotate_operator(operator_account: address) acquires T {
```

Comment:
We should! I think this will be my next PR! The challenge here is that most of the tests now assume that "validator account" == "validator_operator account", and it's not possible to instantiate both roles under a single account, so I essentially don't instantiate the validator operator role now anywhere...

Reply:
even if we perform this check, it isn't guaranteed that the operator is still set afterward, so we'll need to check this along with valid every reconfiguration.